modules/internal/ChapelDomain.chpl

```diff
@@ -378,21 +378,31 @@ module ChapelDomain {
                   "' using count(s) of type ", counts.type:string);
   }
 
+  //
+  // Disallow additions and subtractions to rectangular domains of these types
+  // with a specific message, to avoid surpises.
+  // Note: add/sub of a rectangular domain and another type will either
+  // produce a generic error message or compile as a promoted expression.
+  //
+  private proc noRDadds(type t) param do return
+    isPrimitive(t) || isRange(t) || isTuple(t) || isEnum(t);
+
   @chpldoc.nodoc
-  operator +(d: domain, i: index(d)) {
-    if d.isRectangular() then
-      compilerError("Cannot add indices to a rectangular domain");
-    else
-      compilerError("Cannot add indices to this domain type");
```

Comment:
This is not present after this PR, but appears dead (after out-of-band discussion). It would be great if this was mentioned in the OP.